### ISSUE TITLE
Add Persian Collate to Odoo Database

### ISSUE THREAD

**Issue by AryanFanid** [odoo/odoo#96573]
Impacted versions: 14.0, 15.0

Steps to reproduce: When ordering a column in database, some characters like 'پ' are appearing last.

Current behavior: Default collate for Persian language in DB is Arabic.

Expected behavior: Database order based on characters must be correct and based on Persian(Farsi) collate.

Video/Screenshot link (optional):

Support ticket number submitted via odoo.com/help (optional):

**Comment by mrbayati2002** [comment]
Hi,
the collation of odoo databases is not based on odoo user language.
According to https://github.com/odoo/odoo/blob/15.0/odoo/service/db.py#L111 the default collate is `C` for created database from odoo.

**Comment by AryanFanid** [comment]
Hi,
@mrbayati2002 thanks for your reply. Your info is correct but i have issue when sorting columns when the data is in Persian language. The sorting is not correct!

**Comment by mrbayati2002** [comment]
Yes, you are right. It's a problem to sorting data, but in my opinion there is no other way. A fixed collation must be set in database creation and it is not depend on user language.
Maybe there is a way to set collation on `select` command and use custom queries in model and same way using direct queries on database.
thanks

**Comment by mrbayati2002** [comment]
example of using collation in select queries: https://www.postgresql.org/docs/current/collation.html
If you find another solution, I appreciate it to share your solution.

**Comment by gdgellatly** [comment]
C collation is only used for database created using template0. If you want a different template use a different template. Generally a localized install will have a template1 with the local collation. It is just a config setting.

Just beware text searching and ordering performance will suffer not using C collation.
